docs/ARCHITECHTURE.md

```diff
@@ -0,0 +1,46 @@
+# Batarangle Architecture
+
+The architecture of the Batarangle dev tools extensions can be broken down into 3 parts:
+
+1. **Front-end**: display of block/tree view of the inspected application and other debugging information.
```

Comment:
> block/tree view

We should be careful using these terms without defining them first. It's confusing enough given that the block view is an alternate view of the component tree. I like terms `component` and `performance` views better.

/cc @winkerVSbecks Any preference?